llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -3335,12 +3335,21 @@ bool SimplifyCFGOpt::speculativelyExecuteBB(BranchInst *BI,
     assert(!getLoadStoreType(I)->isVectorTy() && "not implemented");
     auto *Op0 = I->getOperand(0);
     Instruction *MaskedLoadStore = nullptr;
+    PHINode *PN = nullptr;
     if (auto *LI = dyn_cast<LoadInst>(I)) {
       // Handle Load.
       auto *Ty = I->getType();
-      MaskedLoadStore = Builder.CreateMaskedLoad(FixedVectorType::get(Ty, 1),
-                                                 Op0, LI->getAlign(), Mask);
-      I->replaceAllUsesWith(Builder.CreateBitCast(MaskedLoadStore, Ty));
+      Value *PassThru = nullptr;
+      if (I->hasOneUse())
+        if ((PN = dyn_cast<PHINode>(I->use_begin()->getUser())))
+          PassThru = Builder.CreateBitCast(PN->getIncomingValueForBlock(BB),
```

Comment:
Can you please fix the header comment?
https://github.com/llvm/llvm-project/blob/43f044ba7ea2cd668805c01055eb2dff70532ba3/llvm/lib/Transforms/Utils/SimplifyCFG.cpp#L3043

It should be `///     %phi = phi [ %sub, %ThenBB ], [ 0, %BB ] `.

Reply:
Done.